### PR TITLE
WNPRC Test Refactor Updates

### PR DESCRIPTION
**Pull request by labkey-martyp**
#### Rationale
We recently refactored the WNPRC tests to enable the WNPRC modules at the beginning of test setup and before importing the study. This is consistent with how other PRCs are testing. This will ensure triggers and other business logic is exercised and validated with sample data and other project setup. A previous update was merged in with test failures still in the WNPRC repo. This clears up those test failures.

#### Related Pull Requests
* https://github.com/LabKey/wnprc-modules/pull/535

#### Changes
* Test data updates
* Add behavior abstract dataset
* Many test initialization updates
* Update some test animal Ids to match species assignment in WNPRC triggers
* Update primateid trigger to handle import test data
* Update WNPRC billing test setup and test data

### CODE REVIEW

File: primateid/resources/scripts/primateid/triggers.js [hunk 1]
```diff
@@ -68,11 +68,18 @@ exports.init = function (EHR) {
             filterArray: [LABKEY.Filter.create('participantid', row['participantid'], LABKEY.Filter.Types.EQUAL)],
             requiredVersion: 17.1,
             success: function (data) {
+
+                // Getting the participantId field is the normal case, but in auto-tests we need to grab the Id field
+                var participantid = row['participantid'];
+                if (!participantid) {
+                    participantid = row['Id'];
+                }
```
Comment (labkey-bpatel) [inline primateid/resources/scripts/primateid/triggers.js:76]:
```suggestion
var participantid = row['participantid'] || row['Id'];
```

File: WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java [hunk 2]
```diff
@@ -3280,13 +3432,13 @@ private void deathNotificationSetup() throws UnhandledAlertException {
...
-        clickAndWait(Locator.tagWithAttributeContaining("a", "id", "org.labkey.wnprc_ehr.notification.DeathNotificationRevamp").withText("Manage Subscribed Users/Groups"), 0);
-        log("Attempting to select combo box item: EHR Administrators");
-        _ext4Helper.selectComboBoxItem("Add User Or Group:", "EHR Administrators");
-        log("Successfully selected combo box item: EHR Administrators");
-        clickButton("Close", 0);
+//        clickAndWait(Locator.tagWithAttributeContaining("a", "id", "org.labkey.wnprc_ehr.notification.DeathNotificationRevamp").withText("Manage Subscribed Users/Groups"), 0);
+//        log("Attempting to select combo box item: EHR Administrators");
+//        _ext4Helper.selectComboBoxItem("Add User Or Group:", "EHR Administrators");
+//        log("Successfully selected combo box item: EHR Administrators");
+//        clickButton("Close", 0);
```
Comment (labkey-bpatel) [inline WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java]: Does this need to be uncommented?

File: primateid/resources/scripts/primateid/triggers.js (hunk 1 above)
Comment (labkey-bpatel) [inline primateid/resources/scripts/primateid/triggers.js:76]: var participantid = row['participantid'] || row['Id'];

**Review by labkey-bpatel — APPROVED** [review]

File: WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java (hunk 2 above)
Comment (labkey-martyp) [reply WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java]: Looks like Sweta refactored this to a single line. I deleted this commented code.

File: primateid/resources/scripts/primateid/triggers.js (hunk 1 above)
Comment (labkey-martyp) [reply primateid/resources/scripts/primateid/triggers.js:76]: These triggers use really old JS. I'm not sure it even recognizes || in assignment. Just to be safe I'm going to leave this.

**Pull request merged**